tests/generate_bg_Lexical_Analyzer_pass_fail.py

```diff
@@ -146,24 +152,60 @@ def _in_splits(s, splits):
         return s in [list(map(str, ss)) for ss in splits]
 
     f = bg_refs[0]
-    s = [re.sub('H$', 's', SanskritObject(sx, encoding=ITRANS).canonical()) for sx in bg_refs[1]]
-    i = SanskritObject(f, encoding=ITRANS)
     try:
-        # for sss in s:
-        #   if not lexan.forms.valid(sss):
-        #       return "Skip"
-        graph = lexan.getSandhiSplits(i)
+        graph = lexan.parse(f)
+        #print(graph.serializable())
         if graph is None:
-            logger.error("FAIL: Empty split for {}".format(i.canonical().encode('utf-8')))
+            logger.error("FAIL: Empty split for {}".format(bg_refs[0].encode('utf-8')))
             return False
-        # Reducing max_paths to 100
-        splits = graph.findAllPaths(max_paths=100, sort=False)
-        r = _in_splits(s, splits)
-        if splits is None or (not r):
-            logger.error("FAIL: {} not in {}".format(s, splits))
-        return r
-    except:  # noqa
-        logger.warning("Split Exception: {}".format(i.canonical().encode('utf-8')))
+        if graph is None or graph is False:
+            return "Error"
+
+        # make two copies of split for further processing
+        parseSplit = bg_refs[1]
+        expectedSplit = bg_refs[1]
+        indexValue = -1
+        for splitIndex, splitString in enumerate(graph.splits(max_splits=10)):
+                didNotFind = False
+                splitString = splitString.__str__().strip('][').split(', ')
+
+                # Convert input split into SLP format since parser returns in slp format
+                slpSplits = []
+                for itransStr in parseSplit:
+                    slpStr = transliterate(itransStr,sanscript.ITRANS, sanscript.SLP1 )
+                    slpSplits.append(slpStr)
+
+                # remove unwanted " ' " from Split String
+                cleanedSplits = []
+                for splitKhanda in splitString:
+                    splitKhandaClean = splitKhanda.strip("'")
+                    cleanedSplits.append(splitKhandaClean)
+                
+                # check if the cleaned split is present in slpSplits.
+                # if not found, set didNotFind to True which is used later for sandhi
+                # failure check
+                split = cleanedSplits
+                for slpSplit in slpSplits:
+                    if (slpSplit not in split):
+                        didNotFind = True
```

Comment:
This loop may miss the case where a word in the split occurs twice, but is present only once in the generated splits.